src/spineml/simulator/simulator.cc

```diff
@@ -0,0 +1,820 @@
+                                                            const std::map<std::string, std::string> &componentURLs,
+                                                            const ComponentEventPorts &componentEventPorts)
+{
+    const unsigned long long numTimeSteps = calcNumTimesteps();
+    // Get name of target
+    std::string target = node.attribute("target").value();
+
+    // Find size of target component
+    auto targetSize = componentSizes.find(target);
+    if(targetSize == componentSizes.end()) {
+        throw std::runtime_error("Cannot find component '" + target + "'");
+    }
+
+    // If this writing to file or network
+    const std::string hostName = node.attribute("host").value();
+    const bool shouldLogToFile = hostName.empty();
+
+    // If target is an event send port
+    const std::string port = node.attribute("port").value();
+    if(isEventSendPort(target, port, componentURLs, componentEventPorts)) {
+        // **TODO** spike-based network logging not supported
+        assert(shouldLogToFile);
+
+        // Get host and device (if applicable) pointers to spike counts, spikes and queue
+        unsigned int *hostSpikeCount;
+        unsigned int *hostSpikes;
+        unsigned int *spikeQueuePtr;
+        VoidFunction pushFunc;
+        VoidFunction pullFunc;
+        std::tie(hostSpikeCount, hostSpikes, spikeQueuePtr, pushFunc, pullFunc) = getNeuronPopSpikeVars(SpineMLUtils::getSafeName(target));
+
+        // Create event logger
+        return std::unique_ptr<LogOutput::Base>(new LogOutput::Event(node, getDT(), numTimeSteps, port, targetSize->second,
+                                                                     logPath, spikeQueuePtr,
+                                                                     hostSpikeCount, hostSpikes, pullFunc));
+    }
+    // Otherwise we assume it's an analogue send port
+    else {
+        // If there is a dictionary of properties for target population
+        auto targetProperties = m_ComponentProperties.find(target);
+        if(targetProperties != m_ComponentProperties.end()) {
+            // If there is a model property object for this port return an analogue log output to read it
+            auto portProperty = targetProperties->second.find(port);
+            if(portProperty != targetProperties->second.end()) {
+                if(shouldLogToFile) {
+                    return std::unique_ptr<LogOutput::Base>(new LogOutput::AnalogueFile(node, getDT(), numTimeSteps, port, targetSize->second,
+                                                                                        logPath, portProperty->second.get()));
+                }
+                else if(hostName == "0.0.0.0") {
+                    // Create logger
+                    LogOutput::AnalogueExternal *log = new LogOutput::AnalogueExternal(node, getDT(), port, targetSize->second,
```

Comment:
As before, I'd declare the ``unique_ptr`` here to prevent a possible memory leak

Reply:
Done!